tests/data_context/store/test_datasource_store.py

```diff
@@ -222,3 +306,78 @@ def test_datasource_store_update_raises_error_if_datasource_doesnt_exist(
         )
 
     assert f"Unable to load datasource `{datasource_name}`" in str(e.value)
+
+
+def test_datasource_store_with_inline_store_backend_config_with_names_does_not_store_datasource_name(
+    datasource_config_with_names: DatasourceConfig,
+    datasource_config: DatasourceConfig,
+    empty_data_context: DataContext,
```

Comment:
Can we please mark with `integration` due to file I/O?

Reply:
Good call. Hope to eventually update our standard fixtures to use the in memory filesystem.